extensions/ql-vscode/src/pure/messages.ts

```diff
@@ -499,14 +537,46 @@ export interface CompiledUpgrades {
    * The path to the new dataset statistics
    */
   newStatsPath: string;
+  /**
+   * The sha of the target dataset.
+   */
+  targetSha: string;
+}
+
+
+/**
+ * A compiled upgrade. 
+ * The upgrade is spread among multiple files.
+ */
+interface MultiFileCompiledUpgrades extends CompiledUpgradesBase {
+  /**
+   * The path to the new dataset dbscheme
+   */
+  newDbscheme: string;
   /**
    * The steps in the upgrade path
    */
   scripts: CompiledUpgradeScript[];
   /**
-   * The sha of the target dataset.
+   * Will never exist in an old result
    */
-  targetSha: string;
+  compiledUpgradeFile?: never;
+}
+
+
+/**
+ * A compiled upgrade. 
+ * The upgrade is in a single file.
+ */
+export interface SingleFileCompiledUpgrade {
+  /**
+   * The steps in the upgrade path
+   */
+  descriptions: UpgradeDescription[];
+  /**
+   * The file containing the upgrade
```

Comment:
For consistency, keep the comments the same, so we know they are the same thing. If they aren't the same thing, then be explicit about it.

```suggestion
   * A uri pointing to the compiled upgrade file.
```